content/blog/2022/2022-08-03-feature-flags-in-nodejs-backend-server-with-launchdarkly.md

```diff
@@ -0,0 +1,720 @@
+		+ ` known IP addresses.` ) ); 
+	} else { 
+		console.log( chalk.green( 'Allowing IP', chalk.bold( request.ip ) ) ); 
+	} 
+}
+```
+
+Next we can define the script as part of `package.json`:
+
+```json
+{
+	"scripts": {
+    	"blacklist": "nodemon blacklist_check.js"
+  	}
+}
+```
+
+Then we can execute the following command to run our app:
+
+```bash
+npm run blacklist
+```
+
+Finally when we run the above command it will print something like below:
+
+```bash
+info: [LaunchDarkly] Initializing stream processor to receive feature flag updates
+info: [LaunchDarkly] Opened LaunchDarkly stream connection
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+```
+
+Now we can go to the UI, add this IP and redeploy the variation. We will immediately see dynamically the app has picked the blocked IP and start printing it out something like below:
+
+```bash
+info: [LaunchDarkly] Initializing stream processor to receive feature flag updates
+info: [LaunchDarkly] Opened LaunchDarkly stream connection
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Allowing IP 192.168.100.40
+Blocking IP 192.168.100.40
+... blocking one of 2 known IP addresses.
+Blocking IP 192.168.100.40
+... blocking one of 2 known IP addresses.
+Blocking IP 192.168.100.40
+... blocking one of 2 known IP addresses.
+Blocking IP 192.168.100.40
+... blocking one of 2 known IP addresses.
+Blocking IP 192.168.100.40
+... blocking one of 2 known IP addresses.
+Blocking IP 192.168.100.40
+... blocking one of 2 known IP addresses.
+```
+
+So this helped us to minimistically add an administrative feature to our existing microservice.
+
+### Update Batch Size while Performing Batch Operations on the Fly
+
+Consider that you have a batch task or job that needs to be executed. But we might need to update or somehow ramp-up the number of operations as per the need. So, if we think about it, we might need to first run this job behind some kind of scheduler and then we might need to pass some environment variables. Now if we want to update that number we always would need to restart our application.
+
+But we can just make use of our multivariate options in LaunchDarkly to define a batch size. We can dynamically change this values so that it reflects in the code accordingly.
+
+{{% image alt="Batch Multivariate option" src="images/posts/nodejs-backend-launchdarkly/Batch_Size_Count.png" %}}
+
+Next we can define a `BatchRunner` to get batch size and execute the batch process in a loop:
+
+```javascript
+import chalk from 'chalk';
+import LaunchDarkly from 'launchdarkly-node-server-sdk';
+ 
+class BatchRunner {
+ 
+	constructor( launchDarklyClient, flagKey, buyerKey ) { 
+		this._launchDarklyClient = launchDarklyClient;
+		this._flagKey = flagKey;
+		this._buyerKey = buyerKey;
```

Comment:
Again, I don't think we need a user key at all for global feature flags. Instead, we can pass a static `global` or something as the user key, because this feature is not based on a real user.